### DISCUSSION TITLE
A couple of odd issues since the August 3rd Release. The job text is now fixed. Smoothed heart rate line is broken.

### DISCUSSION THREAD

**Discussion by gitanO0** [NightscoutFoundation/xDrip#3626] [Bug reports]
As instructed through the below discussion post:
https://github.com/NightscoutFoundation/xDrip/discussions/2403

**_Discussion 2403 content
Since the August 3rd release of xdrip+ I've been seeing a couple of minor odd issues. I've loaded all the nightlies after August 3rd and these issues are there. If I reload August 3rd, the issues go away.

The IoB text at the top of the main screen, just under the xDrip+ title, is very flaky, sometimes it's there but most of the time that text just says insulin xxx or carbs yyy but as of the August 3 release and earlier there should also always be an IoB text... assuming you actually have insulin on board.

I have the heart and steps functionality enabled through communications with my Fitbit and again, at or before the august 3 release, the heartbeat graph data was a nice smooth contiguous line. In the latest releases, that line is broken up and discontinuous. I've gone back and forth between builds and this seems to be a by-product of the builds, not the input data. Have there been any reports of similar issues? If so, any word on what's going on?
End Discussion 2403 content_**

Both of the below screenshots were taken within minutes of each release being loaded on my Samsung Galaxy S20 Android 12 Device.  I've not changed any settings between installs.  When I load August 3rd release back on, the issues revert to the proper "normal" state.

I've enabled the steps and heartbeat graph data via the Smartwatch Sensors section in Settings.  I also have the Inter-app settings for "xdrip web service" enabled so I can get and push xdrip data to/from my Fitbit Sense.  This functionality has been working just fine for a long time up to the August 3rd release.

**Screenshot 1:  Nov 5 Nightly release:  No IOB text at top of the screen.  Discontiguous heartbeat graph line.  Slightly smaller (light aqua blue) "Steps" graphics.** 
![Screenshot_2022_11_05_23_44_34](https://user-images.githubusercontent.com/6887761/200214323-3be282db-59c2-45d9-8ede-36d588c68590.jpg)

**Screenshot 2:  August 3 Nightly release:  IOB text is present at the top of the screen.  The heartbeat graph line is contiguous.  Larger (normal) "steps" graphics.**
![Screenshot_2022_11_05_23_47_57](https://user-images.githubusercontent.com/6887761/200214327-cabf625b-9113-423c-8aa4-caaed9a44490.jpg)

**Comment by emp-00** [comment]
> Nov 5 Nightly release: No IOB text at top of the screen.

Confirmed: Yesterday, I updated from a ~4 months-old nightly (2022-06-20) to the 2022-11-08 revision. Ever since my IoB display is "broken".

Symptom: When opening xDrip+ in the first 1-2 minutes the IoB display is vanished and only appears after this time-span. This is always the case and clearly linked to the update. Obviously a bug from my point-of-view. This is quite annoying, because I need to have a quick look at my IoB and now I need to wait until it's appearing again. See screenshot for what's happening. Thanks for looking into this. I have to revert to my last known good nightly.

![image](https://user-images.githubusercontent.com/60714349/202553862-36455834-fec9-4a8b-9774-250687603e47.png)

**Comment by gitanO0** [comment]
It looks like the heart rate graph issue is related to the "smooth heart rate graph" option being checked.  Unchecking that option clears the issue up but creates a very jagged heartbeat line.

**Comment by zwidow** [comment]
Updating, bug still present in the last beta released.
Last beta version 20221116 was released with the IoB bug not solved and substitute a version (beta 20220803 where IoB working) 
Since IoB is really important for ppl without pump like me I've been forced to install the old beta apk.

**Comment by gitanO0** [comment]
I can confirm as well that the latest 12/5 nightly retains the above two bugs and for me as well, I can't update past the 08/03 release because of the IoB issue.

**Comment by emp-00** [comment]
Confirmed from my side as well - current nightlys are still unusable.

**Comment by jamorham** [comment]
Please try the Dec 9th Nightly (available now) and let me know if it resolves the issue? Thanks

**Comment by zwidow** [comment]
Tested the last (Dec 9th Nightly) and seems good @jamorham 
Thank you

**Comment by emp-00** [comment]
Confirmed as well - problem seems to be fixed @jamorham ! Furthermore, when adding insulin treatments the IoB figure get's updated _instantly_ (before this, it always took a few seconds) --> me = very happy **Thanks**

**Comment by gitanO0** [comment]
@jamorham The IoB text seems to be staying visible correctly but the heart rate graph is still wonky.  See my initial screenshot above.  There are all kinds of gaps in the heartrate line on the graph.  Also, lesser of an issue... the "steps" graphics (the aqua blue graphics) in the graph are somewhat smaller than on the 8/3 release making them harder to see when there are just a few steps.  When I'm out running and there are lots of steps, it's not so bad.

**Comment by gitanO0** [comment]
@jamorham here's another screenshot.  The heart rate line is wonky and the steps are really small.  See the circled elements.
![Screenshot_20221209-123520](https://user-images.githubusercontent.com/6887761/206783416-ac1cc171-302b-4aa2-9b28-408233374cdb.png)

**Comment by zwidow** [comment]
mine is ok
![image](https://user-images.githubusercontent.com/80387707/206787944-5f6bbb1d-93d7-4cf0-ac70-18559b337f5c.png)

**Comment by jamorham** [comment]
@gitanO0 can you describe the source for your heart rate and steps data?

**Comment by gitanO0** [comment]
@zwidow try turning on the "smooth heart rate graph" option in the "smartwatch sensors" area.  Mine looks good if that option is unchecked but when I check that option, it has the above results.   I like the look of the smoothed line better but will uncheck that option if I have to.

@jamorham I'm getting the data from my Fitbit via local web service interaction... I believe.

**Comment by zwidow** [comment]
can confirm the issue is related to the "smooth heart rate graph" ON

**Comment by machiyag** [comment]
I reverted to a previous version when the IoB reading disappeared, and have been getting reminders to update ever since. Is the consensus that it's now safe to do so?

**Comment by Navid200** [comment]
@machiyag If you are on the stable, beta or alpha update channel, updating now will return the problem for you.

If you are on the Nightly channel, yes, updating will be OK.  

Alternatively, you can stay on the channel you have been on before but, manually install the current nightly:  https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.12.09

**Comment by machiyag** [comment]
Thanks, I'm set to get the stable. Looking at the last .apk in my
downloads, I think I was running the 11/16 build. I've just replaced it
with the 12.09.

At some point I'll be authorized for the Omnipod 5 looping with the G6, and
I expect that to introduce a new set of problems. I've always
double-entered insulin and carbs, but obviously won't be able to do that
with a constant insulin drip. Hoping the benefit outweighs the grief.

Thanks again,

mh

On Sat, Dec 24, 2022, 5:28 AM Navid ***@***.***> wrote:

> @machiyag <https://github.com/machiyag> If you are on the stable, beta or
> alpha update channel, updating now will return the problem for you.
>
> If you are on the Nightly channel, yes, updating will be OK.
>
> Alternatively, you can stay on the channel you have been on before but,
> manually install the current nightly:
> https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.12.09
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NightscoutFoundation/xDrip/issues/2435#issuecomment-1364530410>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKOO6V6HFPHNWD6CSXIO3SLWO33ADANCNFSM6AAAAAARYXU4MI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
mh

**Comment by Navid200** [comment]
@gitanO0 
Please edit the title to be clear what the issue is.  In this case, hear rate.

**Comment by gitanO0** [comment]
Done

**Comment by Son1k02** [comment]
Currently experiencing this exact problem since updating from 20240807-f96d4e5

![Screenshot_20240912_110640_xDrip+](https://github.com/user-attachments/assets/af1e50a0-3536-4ac8-9aa7-db4c4b6309c2)
![Screenshot_20240912_110620_xDrip+](https://github.com/user-attachments/assets/caf20141-3878-4f57-871c-a0ed288e2cc4)
